Lorem Ipsum XR Translator/Assets/Scripts/SpeechToText/SpeechToTextDisplay.cs

```diff
@@ -0,0 +1,177 @@
+using UnityEngine;
+using Microsoft.MixedReality.Toolkit.UI;
+using System.Collections;
+using Translator;
+using Config;  
+
+namespace STT
+{
+    /// <summary>
+    /// The class SpeechToTextDisplay is defined within the STT namespace is used to translate and display in the scene 
+    /// </summary>
+    public class SpeechToTextDisplay : MonoBehaviour
+    {
+        /// <summary>
+        /// Calls the AzureSpeechToText script
+        /// </summary>
+        [SerializeField]
+        private AzureSpeechToText azureSpeechToText; 
+
+        
+        /// <summary>
+        /// originalToolTip is used to display the text converted from Speech to text
+        /// </summary>
+        [SerializeField]
+        private ToolTip originalToolTip; 
+
+        
+        /// <summary>
+        /// translatedToolTip is used to display the translated text
+        /// </summary>
+        [SerializeField]
+        private ToolTip translatedToolTip; 
+
+        /// <summary>
+        /// Azure Translator setup
+        /// </summary>
+        private AzureTranslator azureTranslator;
+        private string apiKey;  
+        private const string location = "northeurope";
+        private const string targetLanguage = "es"; 
+
+        private void Awake()
+        {
+            apiKey = Secrets.GetAzureTranslatorKey();  
+            azureTranslator = new AzureTranslator(apiKey, location);
+        }
+
+        
+        /// <summary>
+        /// This method is called before the first frame update in Unity. It checks if the tooltips are assigned and sets them to be inactive (hidden) initially.
+        /// </summary>
+        private void Start()
+        {
+            if (originalToolTip != null)
+            {
+                originalToolTip.gameObject.SetActive(false);
+            }
+            else
+            {
+                Debug.LogWarning("Original ToolTip is not assigned.");
+            }
+
+            if (translatedToolTip != null)
+            {
+                translatedToolTip.gameObject.SetActive(false);
+            }
+            else
+            {
+                Debug.LogWarning("Translated ToolTip is not assigned.");
+            }
+        }
+
+         /// <summary>
+        /// TranslateAndDisplay which is a asynchronous method is called when a user action (button press) occurs.
+        /// It requests Azure's Speech to Text service and translation service to recognize speech and then displays the recognized and translated text in the tooltips.
+        /// </summary>
+        public async void OnRequestSpeechToText()
+        {
+            
+            /// <summary>
+            /// Call the SSTRequest method and get the recognized text or error message
+            /// </summary>
```

Comment:
Inline comments don't need the /// format. Use /// for functions and classes (and maybe the occasional variable that needs extra explaining). Otherwise // will suffice without the <summary> tags. This doesn't need to change for my approval, but just something to keep in mind going forward. (I do prefer/// to no comments though for sure!)